pkg/utils/utils_test.go

```diff
@@ -230,3 +232,69 @@ func TestRetrieveParamsFromBSL(t *testing.T) {
 		logger.Infof("Repository Parameters: %v", repositoryParameters)
 	}
 }
+
+func TestRerieveVcConfigSecret(t *testing.T) {
+	// Setup Logger
+	logger := logrus.New()
+	formatter := new(logrus.TextFormatter)
+	formatter.TimestampFormat = time.RFC3339Nano
+	formatter.FullTimestamp = true
+	logger.SetFormatter(formatter)
+	logger.SetLevel(logrus.DebugLevel)
+
+	tests := []struct{
+		name string
+		sEnc string
+		vc string
+		password string
+	}{
+		{
+			name: "Password with special character \\ in it",
+			sEnc : "[VirtualCenter \"sc-rdops-vm06-dhcp-184-231.eng.vmware.com\"]\npassword = \"GpI4G`OK'?in40Fo/0\\\\;\"",
+			vc: "sc-rdops-vm06-dhcp-184-231.eng.vmware.com",
+			password: "GpI4G`OK'?in40Fo/0\\;",
+		},
+		{
+			name: "Password with multiple = in it",
+			sEnc : "[VirtualCenter \"sc-rdops-vm06-dhcp-184-231.eng.vmware.com\"]\npassword = \"GpI4G`OK'?in40Fo/0\\\\;=h=\"",
+			vc: "sc-rdops-vm06-dhcp-184-231.eng.vmware.com",
+			password: "GpI4G`OK'?in40Fo/0\\;=h=",
+		},
+		{
+			name: "Password with special character \\t in it",
+			sEnc : "[VirtualCenter \"sc-rdops-vm06-dhcp-184-231.eng.vmware.com\"]\npassword = \"G4\\t4t\"",
+			vc: "sc-rdops-vm06-dhcp-184-231.eng.vmware.com",
+			password: "G4\t4t",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			lines := strings.Split(test.sEnc, "\n")
+			params := make(map[string]string)
+			parseLines(lines, params, logger)
+			assert.Equal(t, test.vc, params["VirtualCenter"])
+			assert.Equal(t, test.password, params["password"])
+		})
+	}
+}
+
+func parseLines(lines []string, params map[string]string, logger logrus.FieldLogger) {
```

Comment:
Why implement the function again? It is already there in the `RetrieveVcConfigSecret` in pkg/utils/utils.go. 

What about extract that part out from RetrieveVcConfigSecret function and make it an independent function? So that, both RetrieveVcConfigSecret function in the production code and TestRetrieveVcConfigSecret function in the test code can consume it.